spyder/widgets/variableexplorer/tests/test_dataframeeditor.py

```diff
@@ -174,6 +179,36 @@ def test_change_format_with_format_not_starting_with_percent(qtbot, monkeypatch)
     with qtbot.assertNotEmitted(editor.sig_option_changed):
         editor.change_format()
 
+def test_header_encoding():
+    data =  u"""a,Unieke_Idcode,ü,b,1
+            0,flooooo,0.55,10100103,31/12/2016
+            """
+    test_data = StringIO(data)
+    df = read_csv(test_data, encoding='utf-8')
+    editor = DataFrameEditor(None)
+    editor.setup_and_check(df)
+    model = editor.dataModel
+    assert model.headerData(0, orientation=Qt.Horizontal) == u"Index"
+    assert model.headerData(1, orientation=Qt.Horizontal) == u"a"
+    assert model.headerData(2, orientation=Qt.Horizontal) == u"Unieke_Idcode"
+    assert model.headerData(3, orientation=Qt.Horizontal) == u"ü"
+    assert model.headerData(4, orientation=Qt.Horizontal) == u"b"
+    assert model.headerData(5, orientation=Qt.Horizontal) == u"1"
+
+def test_header_bom():
+    df = read_csv('spyder/widgets/variableexplorer/tests/test.csv',
+                  encoding='utf-8-sig')
```

Comment:
Please use for this file the one I'm attaching in this comment and call it `issue_2514.csv`. Also

1. There's no need to add the entire path of the file because `read_csv` tries to load it first from the current working directory.
2. Again, drop the encoding parameter.

[2514.txt](https://github.com/spyder-ide/spyder/files/680294/2514.txt)